src/clr-angular/data/datagrid/interfaces/column-state.interface.ts

```diff
@@ -4,9 +4,17 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 import { DatagridColumnChanges } from '../enums/column-changes.enum';
+import { ClrDatagridHideableColumn } from '@clr/angular';
+import { TemplateRef } from '@angular/core';
 
 export interface DatagridColumnState {
   changes?: DatagridColumnChanges[]; // This is an array of change types to update
   width?: number; // This is the width calculated for the column
   strictWidth?: number; // This is the strict width if defined in styles/css
+  isHidden?: boolean; // This is the state of column visibility
```

Comment:
Prefer to use `hidden` to track hidden state.

Then I want a separate property to track if a column can be hidden (which right now is defined by the presence of the column template ref).

Reply:
done